src/Infrastructure/Services/ParticipantService.cs

```diff
@@ -94,19 +94,32 @@ private async Task<string> AdminGetUserAsync(string email)
             CreatedAtUtc = _clock.Now(),
         };
         // check if demographic data is complete
-        var demographics = await _participantRepository
+        var oldUser = await _participantRepository
             .GetParticipantDemographicsAsync(participant.Pk.Replace("PARTICIPANT#", ""));
-        if (demographics.HasDemographics)
+        if (oldUser.HasDemographics)
         {
             await _participantRepository.CreateParticipantDetailsAsync(entity);
-            await _participantRepository.AddDemographicsToNhsUserAsync(demographics, entity.NhsId);
+            await _participantRepository.AddDemographicsToNhsUserAsync(oldUser, entity.NhsId);
         }
         else if (participant.ConsentRegistration)
         {
             // create new user with consent and deactivate the old user
             await _participantRepository.CreateParticipantDetailsAsync(entity);
         }
 
+        // get the linked account row by passing the ParticipantId into the pk as the new row saves the PK as ParticipantId to link 
+        var linkedRecord =
+            await _participantRepository.GetParticipantAsync(
+                $"PARTICIPANT#{oldUser.ParticipantId}");
+
+        // delete the linked row
+        await _participantRepository.DeleteParticipantAsync(linkedRecord);
+
```

Comment:
Consider what the worst case scenario is in this case.

How 'bad' would it be if the code fails here?
The old record has been deleted before the new linked record is created.
Is that recoverable?

Would it be better to create the new record before deleting the old one?
Is that better or worse if the process fails between those two operations?

Reply:
This is for reporting purposes only so nothing really application critical.  I have moved to the end of the file though as there is more important logic before this.  I have also moved the creation before the deletion.